app/scripts/modules/core/src/application/modal/ApplicationProviderFields.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { isEmpty, isString } from 'lodash';
+
+import { Application } from 'core/application';
+import { CloudProviderRegistry } from 'core/cloudProvider';
+
+export interface IApplicationProviderFieldsProps {
+  application: Application;
+  updateApplication: (applicationFieldPath: string, value: any) => void;
+  cloudProviders: string[];
+}
+
+export function ApplicationProviderFields(props: IApplicationProviderFieldsProps) {
+  const componentPath = 'applicationProviderFields.component';
+
+  const getRelevantProviderFieldsComponent = () => {
+    const appCloudProviders = props.application.cloudProviders;
+    const candidateProvidersToShow: string[] = isEmpty(appCloudProviders)
+      ? props.cloudProviders
+      : isString(appCloudProviders)
+      ? appCloudProviders.split(',')
+      : appCloudProviders;
+
+    const components: any[] = (candidateProvidersToShow || [])
+      .filter(provider => CloudProviderRegistry.hasValue(provider, componentPath))
+      .map(provider => CloudProviderRegistry.getValue(provider, componentPath));
+    return (
+      <>
+        {components.map((FieldComponent, index) => (
+          <FieldComponent key={index} />
+        ))}
+      </>
+    );
+  };
+
+  return getRelevantProviderFieldsComponent();
```

Comment:
I think it is a bit more readable to separate `const` declaration and the render function. 

```
export function ApplicationProviderFields(props: IApplicationProviderFieldsProps) {
  const componentPath = 'applicationProviderFields.component';
  const appCloudProviders = props.application.cloudProviders;
  const candidateProvidersToShow: string[] = ...   
  const components: any[] = ...

  const getRelevantProviderFieldsComponent = () => (
      <>
        {components.map((FieldComponent, index) => (
          <FieldComponent key={index} />
        ))}
      </>
  );
  return getRelevantProviderFieldsComponent();
};
```

Reply:
👍